### ISSUE TITLE
[BE] 10-10EZ - BUG: Missing Veteran data prohibiting submissions

### ISSUE THREAD

**Issue by hdjustice** [department-of-veterans-affairs/va.gov-team#94779]
## Benefit to the Veteran
As a Veteran, I want to return to my in-progress application, and submit it without errors, so that I can feel confident that my form data is complete and accurate

## Scope
Investigate the issue where an authenticated Veteran is returning to the 10-10EZ form in-progress, and when they try to submit, nothing happens.  It appears that the submit button is not working, the Veteran verifies their data and sometimes leaves the form to reload or return at another time, only for the form to fail again.

In staging, it was found that the error comes up as `Uncaught TypeError: Cannot read properties of undefined (reading 'veteranDateOfBirth')`

### Resources

<details>
<Summary>Video from Staging test (user 16) - Expand to view</Summary>

[See video in comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/94779#issuecomment-2405487979)

</details>

<details>
<Summary>Screenshot of the error - expand to view</Summary>
<img width="1337" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOG82Qnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--de95da36ea84b80ca9ac83de3f213618ca218edf/image.png" alt="image.png" />

</details>

Notes from teammates:
- Looks like the issue is in the helpers/index.js
- logged in and in the state where it is looking there is no `veteranDateOfBirth` key or value as far as I can tell
- Not this thing again…. We have so much in place to attempt to prevent it. There are Cypress tests for it and validation of all sorts for checking the profile value.
- It looks like the value is in there from the profile, just an issue with it being in the submission

## Tasks
- [x] Investigate error (work with Luis and Allan if needed)
- [x] Determine root cause
- [x] Determine solution
- [x] Discuss with team/PM for next steps

## Acceptance criteria
- The issue has been investigated and we have next steps to resolve it.

**Comment by hdjustice** [comment]
https://github.com/user-attachments/assets/009a1432-bc81-4156-bf75-574fa6fd29bd

**Comment by coope93** [comment]
When saving an in progress form, we create an `InProgressForm` in the database with a `form_data` column. This stores the current values in a json object. When a user is authenticated, we get some information about the Veteran from their profile and load it into the `form_data`. It looks like sometimes the `VeteranDateOfBirth` is not set on these, even though we know the profile has it. Before submitting, we validate if we have the `VeteranDateOfBirth`, and if we don't we look to see if we have it stored in another piece of state. The change in the linked PR updated where we set one of the values in state where we are looking for the `VeteranDateOfBirth`, so in this instance it was not in the `form_data`, but when we did the transform for submit it found the `VeteranDateOfBirth` from another place we load it from the profile. 

Like I mentioned in slack, it may be valuable to create a future ticket to investigate why the `VeteranDateOfBirth` isn't always getting saved in the `form_data`, but either way this solution will make it more resilient in case we don't have the value.
